msg/templates/uorb_microcdr/msg.cpp.template

```diff
@@ -141,24 +140,25 @@ def add_code_to_deserialize():
     add_deserialize_functions(sorted_fields, "")
 }@
 
-void serialize_@(topic_name)(const struct @(uorb_struct) *input, char *output, uint32_t *length, struct microCDR *microCDRWriter)
+void serialize_@(topic_name)(MicroBuffer *microCDRWriter, struct @(uorb_struct) *input, char *output, uint32_t *length)
 {
-    if (nullptr == input || nullptr == output || nullptr == length || nullptr == microCDRWriter) return;
+    if (nullptr == microCDRWriter || nullptr == input || nullptr == output || nullptr == length)
+	return;
 
-    resetStaticMicroCDRForSerialize(microCDRWriter);
+    reset_micro_buffer(microCDRWriter);
 
-    serializeUnsignedLong(input->timestamp, microCDRWriter);
+    serialize_uint64_t(microCDRWriter, input->timestamp);
 @add_code_to_serialize()
-
-    (*length) = microCDRWriter->m_microBuffer->m_serializedBuffer;
+    (*length) = (*microCDRWriter->final);
```

Comment:
Assigning the final value to length is a mistake.
`microCDRWriter->final` points one position pass the last serialised element.
Dereferencing it does not gives you the length.
You need to get the distance between init and final pointers:
  `  (*length) = (microCDRWriter->final - microCDRWriter->init);`